File: src/CanPlaceFlowers.java
public class CanPlaceFlowers {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        return flowersCapacity(flowerbed) >= n;
    }

    private int flowersCapacity(int[] flowerbed) {
        int capacity = 0;
        for (int index = 0 ; index < flowerbed.length ; index++) {
            if ((flowerbed[index] != 1) && (index == 0 || flowerbed[index - 1] == 0) && (index == flowerbed.length - 1 || flowerbed[index + 1] == 0)) {
                capacity++;
                flowerbed[index] = 1;
            }
        }
        return capacity;
    }
}

File: python/can_place_flowers.py
from typing import List


class Solution:
    def flowerBedCapacity(self, flowerbed: List[int]) -> int:
        capacity = 0
        for index, value in enumerate(flowerbed):
            if value != 1 and (index == 0 or flowerbed[index - 1] == 0) and (index == len(flowerbed) - 1 or flowerbed[index + 1] == 0):
                capacity += 1
                flowerbed[index] = 1
        return capacity

    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        return self.flowerBedCapacity(flowerbed) >= n
